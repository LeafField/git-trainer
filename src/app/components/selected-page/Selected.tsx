"use client";
import React from "react";
import Selector from "./select/Selector";

const Selected = () => {
  return (
    <main className="min-h-[calc(100vw - 45px)] w-full ">
      <h1 className="text-4xl md:text-5xl font-bold text-white text-center mt-[5%]">
        難易度選択
      </h1>
      <div className="mt-[8%]">
        <Selector title="初級編:GitHub Flow" url="flow" />
        <Selector title="中級編:制作中" url="dummy" />
        <Selector title="上級編:制作中" last={true} url="dummy" />
      </div>
    </main>
  );
};

export default Selected;
