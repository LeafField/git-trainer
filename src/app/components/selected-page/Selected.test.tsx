import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import * as Router from "next/navigation";
import Selected from "./Selected";

jest.mock("next/navigation");

const mockRouter = (mockCallback?: () => {}) => {
  jest.spyOn(Router, "useRouter").mockImplementation(() => ({
    back: jest.fn(),
    push: mockCallback ? mockCallback : jest.fn(),
    forward: jest.fn(),
    refresh: jest.fn(),
    replace: jest.fn(),
    prefetch: jest.fn(),
  }));
};

describe("Selectedの結合テスト", () => {
  it("Selectedが正しくレンダリングされているか", async () => {
    mockRouter();
    const { container } = render(<Selected />);
    expect(screen.getByText("初級編:GitHub Flow")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
