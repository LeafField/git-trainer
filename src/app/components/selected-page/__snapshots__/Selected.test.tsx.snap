// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectedの結合テスト Selectedが正しくレンダリングされているか 1`] = `
<div>
  <main
    class="min-h-[calc(100svh - 45px)] relative w-full"
  >
    <h1
      class="mt-[5%] text-center text-4xl font-bold text-white md:text-5xl"
    >
      難易度選択
    </h1>
    <div
      class="mt-[8%]"
    >
      <a
        class="block border-t   cursor-pointer border-white py-[2.94rem] text-center text-3xl text-white transition-colors hover:text-gray-400 md:text-4xl"
        href="/difficulty/flow"
      >
        初級編:GitHub Flow
      </a>
      <a
        class="block border-t border-b  cursor-pointer border-white py-[2.94rem] text-center text-3xl text-white transition-colors hover:text-gray-400 md:text-4xl"
        href="/difficulty/other"
      >
        中級編:Other Commands
      </a>
    </div>
  </main>
</div>
`;
